main.c

```diff
@@ -30,32 +30,26 @@
 #define OBJ_KEY 260
 #define MAX_OBJECTS 128
 
-#define generate_err(x) {									\
-	if (err && err == -ENOEXEC) {							\
-		printf("WARN: generated btf (%s) is poisoned%s\n",	\
-			x, env.nopoison ? " (deleting)" : "");			\
-		if (env.nopoison)									\
-			unlink(x);										\
-	} else if (err) {										\
-		printf("ERR : failed to generate btf for %s\n", x);	\
-		return 1;											\
-	}														\
+#define generate_err(x) {                                               \
+	if (err) {                                                      \
+		printf("ERR : failed to generate btf for %s\n", x);     \
+		return 1;                                               \
```

Comment:
It's causing some leaks. On L242 it returns without calling `closedir`. I'll propose a solution after this PR is merged.

Reply:
Thanks for checking that!